config/checkstyle-checks.xml

```diff
@@ -485,6 +485,17 @@
                value="'@noinspection' Javadoc tags should be accompanied by a
                      '@noinspectionreason' tag, explaining why we suppressed inspection."/>
     </module>
+    <module name="MatchXpath">
+      <property name="query"
+                value="//CLASS_DEF[./IDENT[ends-with(@text, 'Check') or ends-with(@text, 'Filter')]]
+                //METHOD_DEF[./IDENT[starts-with(@text, 'set')]
```

Comment:
Why any depth on `CLASS_DEF` and `METHOD_DEF`? Won't this also check nested classes unnecessarily?

Reply:
You're right. Done.